test/system/520-checkpoint.bats

```diff
@@ -170,4 +170,38 @@ function teardown() {
 
 # FIXME: test --leave-running
 
+@test "podman checkpoint --file-locks" {
+    run_podman run -d $IMAGE sh -c 'touch test.lock; flock test.lock sh -c "for i in \$(seq 3); do echo \$i; sleep 1; done" & \
+                                                     flock test.lock sh -c "for i in \$(seq 3); do echo \$i; sleep 1; done" & wait'
```

Comment:
I think the test can be optimized.

The sleeping suggests that there is a timing issue.  In theory, it's possible that the container calls `flock` two times before it gets checkpointed. So it looks like a race condition to me.

Couldn't we instead wait for signal?  The container calls `flock`, then waits for a signal, then calls `flock` again.  The `echo` could still be used to be extra sure everything worked as expected.

Reply:
But before changing anything, let's wait for @edsantiago